zerver/lib/actions.py

```diff
@@ -6500,6 +6501,13 @@ def do_remove_realm_domain(
     send_event(realm, event, active_user_ids(realm.id))
 
 
+def do_add_realm_playground(realm: Realm, playground_info: Dict[str, str]) -> int:
```

Comment:
Probably we should call this `check_add_realm_playground` -- we try to reserve `do_` names for things where all input validation is done and we don't expect failures.

Reply:
Oh, I think it's OK but we should add a comment that we expect `full_clean` to always pass because `check_url` does the same validation above.